x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java

```diff
@@ -79,7 +79,7 @@ protected void doMasterOperation(
         ClusterState state,
         ActionListener<ExplainLifecycleResponse> listener
     ) {
-        Map<String, IndexLifecycleExplainResponse> indexResponses = new HashMap<>();
```

Comment:
It's probably worth throwing in a simple test just so that no one accidentally reverts this in the future ("it's faster if we just use a HashMap here, and no tests break")

Reply:
Good idea, I've added a test.